test/rekt/features/channel/features.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package channel
+
+import (
+	"fmt"
+
+	"github.com/cloudevents/sdk-go/v2/test"
+	"knative.dev/eventing/test/rekt/resources/channel_impl"
+	"knative.dev/eventing/test/rekt/resources/containersource"
+	"knative.dev/eventing/test/rekt/resources/delivery"
+	"knative.dev/eventing/test/rekt/resources/pingsource"
+	"knative.dev/eventing/test/rekt/resources/source"
+	"knative.dev/eventing/test/rekt/resources/subscription"
+	"knative.dev/reconciler-test/pkg/eventshub"
+	"knative.dev/reconciler-test/pkg/eventshub/assert"
+	"knative.dev/reconciler-test/pkg/feature"
+	"knative.dev/reconciler-test/resources/svc"
+)
+
+func ChannelChain(length int) *feature.Feature {
+	f := feature.NewFeature()
+	sink := feature.MakeRandomK8sName("sink")
+	cs := feature.MakeRandomK8sName("containersource")
+
+	var channels []string
+	for i := 0; i < length; i++ {
+		name := feature.MakeRandomK8sName(fmt.Sprintf("channel-%04d", i))
+		channels = append(channels, name)
+		f.Setup("install channel", channel_impl.Install(name))
+		f.Requirement("channel is ready", channel_impl.IsReady(name))
+	}
+
+	f.Setup("install sink", eventshub.Install(sink, eventshub.StartReceiver))
+	// attach the first channel to the source
+	f.Setup("install containersource", containersource.Install(cs, pingsource.WithSink(channel_impl.AsRef(channels[0]), "")))
```

Comment:
This seems a bit odd that we're using containersource.Install yet using pingsource.WithSink. I don't think it makes any difference since it's just modding a map inside, but found it odd.